sklearn/cluster/_k_means_elkan.pyx

```diff
@@ -0,0 +1,176 @@
+# cython: cdivision=True
+# cython: boundscheck=False
+# cython: wraparound=False
+#
+# Author: Andreas Mueller
+#
+# Licence: BSD 3 clause
+
+import numpy as np
+cimport numpy as np
+cimport cython
+
+from libc.math cimport sqrt
+
+from ..metrics import euclidean_distances
+from ._k_means import _centers_dense
+
+
+cdef double *getfloatpointer(np.ndarray[np.float64_t, ndim=2, mode='c'] data):
+    return <double *>(data.data)
+
+
+cdef double d(double* a, double* b, int n_features) nogil:
```

Comment:
I don't know for sure, but I think @amueller gave this a short name so that the rest of the code is easier to follow ?